src/Illuminate/Container/Container.php

```diff
@@ -1024,6 +1026,26 @@ protected function resolvePrimitive(ReflectionParameter $parameter)
             return [];
         }
 
+        if ($this->hasConfigRepository()) {
+            foreach ($parameter->getAttributes(InjectedConfig::class) as $attribute) {
+                $attributeName = $attribute->getName();
+
+                if ($attributeName !== InjectedConfig::class) {
+                    continue;
+                }
+
+                if (count($attribute->getArguments()) !== 1) {
+                    throw new \InvalidArgumentException(
+                        'ConfigValueWrapper attribute must have exactly one argument'
+                    );
+                }
```

Comment:
Don't really think so, but I will verify asap

Reply:
When you define the attribute it is a normal PHP Class, in the constructor you can use property promotion to make the `$key` accessible and instead of saying `$attribute->getArguments()` you can say `$attribute->newInstance()` to get an actual instance of the class and then access `$instance->key`. The user is required to supply the key when using the attribute because they're instantiating a new instance of that class and the constructor requires that argument.

Reply:
Thanks! That's new for me. I am updating it immediately